### PR TITLE
Chore optimize docker images to reduce build times

### LINKED ISSUE #291: Optimize docker images to reduce build times
Currently it takes 15 minutes to rebuild our containers in CI even with with Docker's caching. It's likely that we can reorder layers in the images to make caching more effective.

Having properly working cache would also help in cases where a rebuild is not necessary without having to check if the image already exists. If you run `concent-builder/build-test-and-push.yml` again on the same version, the build should be instant.

### PR DESCRIPTION
**Pull request by bartoszbetka**
#291

### CODE REVIEW

File: containers/package-builder/build-concent-api-assets.sh.j2 [hunk 1]
```diff
@@ -10,15 +10,10 @@ cp --recursive "$concent_dir/concent_api" "$build_dir/concent-api-assets"
...
-virtualenv --python "python{{ python_version }}" "$build_dir/virtualenv/"
 source "$build_dir/virtualenv/bin/activate"
-# The requirements needs to be installed from the directory it's in because it contains relative paths.
-cd "$build_dir/repositories/concent/concent_api/"
-python -m pip install -r "requirements.lock"
-
 cp "$package_builder_dir/minimal_settings.py" "concent_api/settings/local_settings.py"
 
-./manage.py collectstatic --noinput
+python -B manage.py collectstatic --noinput
```
[the remark this answers is not in the document]
Comment (bartoszbetka) [reply containers/package-builder/build-concent-api-assets.sh.j2]: I was thinking about the comment, but it is a basic flag that can be checked with `python --help`.

[the remark this answers is not in the document]
Comment (bartoszbetka) [reply containers/package-builder/build-concent-api-assets.sh.j2]: Ok, I add the comment and split this commit.

File: containers/package-builder/build-virtualenv.sh.j2 [hunk 2]
```diff
@@ -9,4 +9,4 @@ virtualenv --python "python{{ python_version }}" "$build_dir/virtualenv/"
 source "$build_dir/virtualenv/bin/activate"
 cd "$build_dir/repositories/concent/concent_api/"
 echo -n "$(git describe --tags)" > ../middleman_protocol/RELEASE-VERSION
-python -m pip install -r "$requirements_path"
+python -m pip install -r  "$requirements_path"
```
[the remark this answers is not in the document]
Comment (bartoszbetka) [reply containers/package-builder/build-virtualenv.sh.j2]: Mistake.

File: containers/package-builder/build-concent-api-assets.sh.j2 [hunk 3]
```diff
@@ -10,12 +10,7 @@ cp --recursive "$concent_dir/concent_api" "$build_dir/concent-api-assets"
 
 cd "$build_dir/concent-api-assets/concent_api/"
 
-virtualenv --python "python{{ python_version }}" "$build_dir/virtualenv/"
 source "$build_dir/virtualenv/bin/activate"
-# The requirements needs to be installed from the directory it's in because it contains relative paths.
-cd "$build_dir/repositories/concent/concent_api/"
```
[the remark this answers is not in the document]
Comment (bartoszbetka) [reply containers/package-builder/build-concent-api-assets.sh.j2]: This scripts from `package-builder` are always only running locally. As I say in the commit name, this lines are executed in the `build-virtualenv.sh` script - they are duplicate and do the same.

[the remark this answers is not in the document]
Comment (cameel) [reply containers/package-builder/build-concent-api-assets.sh.j2]: Sorry, I interpreted the code wrong. This script runs on the host but I thought that it's executed inside a docker container when we build an image. It's OK then.

File: containers/package-builder/build-concent-api-assets.sh.j2 (hunk 1 above)
[the remark this answers is not in the document]
Comment (cameel) [reply containers/package-builder/build-concent-api-assets.sh.j2]: Does not look resolved yet.

[the remark this answers is not in the document]
Comment (bartoszbetka) [reply containers/package-builder/build-concent-api-assets.sh.j2]: I forgot push it, it's ok now.

**Pull request merged**